src/Microsoft.ML.Tokenizers/Utils/StringSpanOrdinalKey.cs

```diff
@@ -45,6 +45,31 @@ public StringSpanOrdinalKey(char* ptr, int length)
         public override int GetHashCode() => Helpers.GetHashCode(Span);
     }
 
+    internal unsafe readonly struct StringSpanOrdinalKeyPair : IEquatable<StringSpanOrdinalKeyPair>
+    {
+        private readonly StringSpanOrdinalKey _left;
+        private readonly StringSpanOrdinalKey _right;
+
+        public StringSpanOrdinalKeyPair(char* ptr1, int length1, char* ptr2, int length2)
+        {
+            _left = new StringSpanOrdinalKey(ptr1, length1);
+            _right = new StringSpanOrdinalKey(ptr2, length2);
+        }
+
+        public StringSpanOrdinalKeyPair(string data1, string data2)
+        {
+            _left = new StringSpanOrdinalKey(data1);
+            _right = new StringSpanOrdinalKey(data2);
+        }
+        public override bool Equals(object? obj) =>
+            obj is StringSpanOrdinalKeyPair wrapper && wrapper._left.Equals(_left) && wrapper._right.Equals(_right);
+
+        public bool Equals(StringSpanOrdinalKeyPair other) => other._left.Equals(_left) && other._right.Equals(_right);
+
+        public override int GetHashCode() => _left.GetHashCode() + _right.GetHashCode();
```

Comment:
Is this a sufficiently-good mechanism for combining these hash codes?

Reply:
I fixed that by using HashCode.Combine